books/free-programming-books-et.md

```diff
@@ -25,24 +31,23 @@
 
 ### Java
 
-* [Java õppematerjalid](https://ained.ttu.ee/javadoc/index.html) - TTÜ õppematerjalid
+* [Java õppematerjalid](https://javadoc.pages.taltech.ee) - TTÜ (Taltech) õppematerjalid
 * [Programmeerimiskeel Java](https://et.wikibooks.org/wiki/Programmeerimiskeel_Java) - Wikiõpikud
 
 
 ### JavaScript
 
-* [JavaScript](http://puhang.tpt.edu.ee/raamatud/JavaScript_konspekt.pdf) - Jüri Puhang (PDF)
-* [Javascript](http://lepo.it.da.ut.ee/~epuman/javascript.html) - Ella Puman, Tartu ÜLikooli õppematerjal
+* [JavaScript](https://web.archive.org/web/20200922201525/http://puhang.tpt.edu.ee/raamatud/JavaScript_konspekt.pdf) - Jüri Puhang (PDF) *(:card_file_box: archived)*
 
 
 #### AngularJS
 
-* [AngularJS raamistiku õppematerjal](http://www.cs.tlu.ee/teemad/get_file.php?id=400) - Tallinna Ülikooli, Sander Leetus (PDF)
+* [AngularJS raamistiku õppematerjal](https://www.cs.tlu.ee/teemad/get_file.php?id=400) - Tallinna Ülikool, Sander Leetus (PDF)
 
 
 #### Vue
 
-* [Vue](http://www.cs.tlu.ee/teemaderegister/get_file.php?id=715) - Tallinna ülikooli õppematerjal
+* [Vue](https://www.cs.tlu.ee/teemaderegister/get_file.php?id=715) - Tallinna Ülikooli õppematerjal
```

Comment:
```suggestion
* [Vue.js raamistiku õppematerjal](https://www.cs.tlu.ee/teemaderegister/get_file.php?id=715) - Tallinna Ülikooli, Fred Korts (PDF)
```

Reply:
We typically credit authors, not publishers. In some other entries, we use "Taltech" as a corporate author, not as the publisher.